src/foam/nanos/dig/SugarWebAgent.java

```diff
@@ -0,0 +1,177 @@
+import foam.lib.json.OutputterMode;
+import foam.nanos.dig.exception.*;
+import foam.nanos.http.HttpParameters;
+import foam.nanos.logger.Logger;
+import foam.util.SafetyUtil;
+import foam.nanos.http.WebAgent;
+
+import java.io.*;
+import java.nio.CharBuffer;
+import java.lang.Exception;
+import java.lang.reflect.*;
+import javax.servlet.http.HttpServletResponse;
+
+public class SugarWebAgent
+  implements WebAgent
+{
+  public SugarWebAgent() {}
+
+  public void execute(X x) {
+    Logger              logger         = (Logger) x.get("logger");
+    PrintWriter         out            = x.get(PrintWriter.class);
+    HttpServletResponse resp           = x.get(HttpServletResponse.class);
+    HttpParameters      p              = x.get(HttpParameters.class);
+    CharBuffer          buffer_        = CharBuffer.allocate(65535);
+    String              serviceName    = p.getParameter("service");
+    String              methodName     = p.getParameter("method");
+    String              interfaceName  = p.getParameter("interfaceName");
+
+    try {
+      if ( SafetyUtil.isEmpty(serviceName) ) {
+        DigErrorMessage error = new EmptyParameterException.Builder(x)
+          .setMessage("Empty Service Key")
+          .build();
+        outputException(x, resp, "JSON", out, error);
+
+        return;
+      }
+
+      if ( SafetyUtil.isEmpty(methodName) ) {
+        DigErrorMessage error = new EmptyParameterException.Builder(x)
+          .setMessage("Empty Method Name")
+          .build();
+        outputException(x, resp, "JSON", out, error);
+
+        return;
+      }
+
+      Class c = Class.forName(interfaceName);
+
+      Method m[] = c.getMethods();  // get Methods' List from the class
+
+      Class[] paramTypes = null; // for picked Method's parameters' types
+      Object arglist[] = null; // to store each parameters' values
+
+      for ( int k = 0 ; k < m.length ; k++ ) {
+        if ( m[k].getName().equals(methodName) ) { //found picked Method
+
+          logger.debug("service : " + serviceName);
+          logger.debug("methodName : " + m[k].getName());
+
+          Parameter[] pArray = m[k].getParameters();
+          paramTypes = new Class[pArray.length];
+          arglist = new Object[pArray.length];
+
+          for ( int j = 0 ; j < pArray.length ; j++ ) {
+            paramTypes[j] = pArray[j].getType();
+
+            if ( ! pArray[j].isNamePresent() ) {
+              DigErrorMessage error = new GeneralException.Builder(x)
+                .setMessage("IllegalArgumentException : Add a compiler argument")
+                .build();
+              outputException(x, resp, "JSON", out, error);
+
+              return;
+            }
+
+            paramTypes[j] = pArray[j].getType();
+            arglist[j] = p.getParameter(pArray[j].getName());
+
+            logger.debug(pArray[j].getName() + " :   " + p.getParameter(pArray[j].getName()));
+
+            // casting and setting according to parameters type
```

Comment:
Switch would be perfect and more efficient/readable for this

Reply:
conditional expressions are different depending on primitive types or not though..?

Reply:
Sorry. I don't understand... But you can have 
`String typeName = pArray[j].getType().getCanonicalName() 
switch(typeName) {
case: "double": 
case: "boolean"
....
}
`

Reply:
In the future, we're going to switch method arguments to FOAM modelled types, the same as for properties. At that point we'll be able to just use the parser provided by the PropertyInfo, which is better than either switch or many if/elses. Until then, a switch would be better than ifelseifelse's. Another method would be to define a static hashmap of String to Parser mappings and then just lookup the parser to use.